clearing_operations/scripts/NumberFeaturesTool.py

```diff
@@ -0,0 +1,310 @@
+            descArea = arcpy.Describe(areaToNumber)
+            areaGeom = descArea.shapeType
+            arcpy.AddMessage("Shape type: " + str(areaGeom))
+            if (descArea.shapeType != "Polygon"):
+                raise Exception("ERROR: The area to number must be a polygon.")
+
+            #Checking the version of the Desktop Application
+            gisVersion = arcpy.GetInstallInfo()["Version"]
+            # global appEnvironment
+            appEnvironment = Utilities.GetApplication()
+            if DEBUG == True: arcpy.AddMessage("App environment: " + appEnvironment)
+
+            #Getting the layer name from the Table of Contents
+            # global mxd
+            # global df
+            # global aprx
+            # global mp
+            # global mapList
+            # mxd, df, aprx, mp = None, None, None, None
+            #if gisVersion == "1.0": #Pro:
+            if appEnvironment == "ARCGIS_PRO":
+                from arcpy import mp
+                aprx = arcpy.mp.ArcGISProject("CURRENT")
+                mapList = aprx.listMaps()[0]
+                for lyr in mapList.listLayers():
+                    if lyr.name == pointFeatureName:
+                        layerExists = True
+            #else:
+            if appEnvironment == "ARCMAP":
+                from arcpy import mapping
+                mxd = arcpy.mapping.MapDocument('CURRENT')
+                df = arcpy.mapping.ListDataFrames(mxd)[0]
+                for lyr in arcpy.mapping.ListLayers(mxd):
+                    if lyr.name == pointFeatureName:
+                        layerExists = True
+
+            if layerExists == False:
+                arcpy.MakeFeatureLayer_management(pointFeatures, pointFeatureName)
+            else:
+                pointFeatureName = pointFeatures
+
+            # Select all the points that are inside of area
+            if areaToNumber:
+                arcpy.AddMessage("Selecting points from {0} inside of the area {1}".format(pointFeatureName, areaToNumber))
+            else:
+                arcpy.AddMessage("Selecting points from {0} inside of the area {1}".format(pointFeatureName, areaToNumber.name))
+
+            selectionLayer = arcpy.SelectLayerByLocation_management(pointFeatureName, "INTERSECT",
+                                                                    areaToNumber, "#", "NEW_SELECTION")
+            if DEBUG == True:
+                arcpy.AddMessage("Selected " + str(arcpy.GetCount_management(pointFeatureName).getOutput(0)) + " points")
+
+            # If no output FC is specified, then set it a temporary one, as this will be copied to the input and then deleted.
+            # Sort layer by upper right across and then down spatially,
+            overwriteFC = False
+            #global outputFeatureClass
+            #if outputFeatureClass == "":
+            if not outputFeatureClass:
+                outputFeatureClass = g_ESRI_variable_1
+                overwriteFC = True;
```

Comment:
Someone misses C programming - remove ";"

Reply:
Ha! Not mine, but will remove. Did C programming only once in college.